### PR TITLE
Support to sign RFQ order

### PR DESCRIPTION
**Pull request by tailingchen**
This PR is to support sign RFQ Order. There is difference between RFQ order and PMM order, describe as below:
Sign order according to different type of maker wallet.

1. EOA:
Use `EthSing` signature type. The signature length is 98, first 65 bytes is  eth_sign signature，the middle 32 bytes is reserved ，and the last byte is signature type.
```
+------|---------|---------|-------------------|------+
|  R   |    S    |    V    | reserved 32 bytes | type |
+------|---------|---------|-------------------|------+
```

2. Contract wallet: 
Use `Wallet` signature type. When verifying signature, the RFQ contract will invoke contract wallet to verify signature.  For maker contract wallet version `v4mmp`, the signature length is 88, first 87 bytes is signature and the last byte is signature type.
```
+------|---------|---------|---------|---------|------+
|  V   |    R    |    S    |userAddr |feeFactor| type |
+------|---------|---------|---------|---------|------+
```

### CODE REVIEW

File: src/handler/newOrder.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { ValidationError } from './errors'
 import { addQuoteIdPrefix, constructQuoteResponse, preprocessQuote } from '../quoting'
 
 import { assetDataUtils, SignedOrder } from '0x-v2-order-utils'
+import { buildSignedOrder as buildRFQV1SignedOrder } from '../signer/rfqv1'
 import { buildSignedOrder } from '../signer/pmmv5'
 import { buildSignedOrder as buildLagacyOrder } from '../signer/pmmv4'
```
Comment (kaichen) [inline src/handler/newOrder.ts:11]: please remove the pmmv4 order related code, it's been deprecated.

File: src/handler/newOrder.ts [hunk 2]
```diff
@@ -121,13 +122,15 @@ function getOrderAndFeeFactor(simpleOrder, rate, tokenList, tokenConfigs, config
   const order = {
     makerAddress: config.mmProxyContractAddress.toLowerCase(),
     makerAssetAmount,
+    makerAssetAddress: makerToken.contractAddress,
     makerAssetData: assetDataUtils.encodeERC20AssetData(
       getWethAddrIfIsEth(makerToken.contractAddress, config)
     ),
```
Comment (kaichen) [inline src/handler/newOrder.ts:128]: keep line L125, and remove this encoding.

only do token address encoding in pmmv5. ammv1 don't need this either.

File: src/signer/rfqv1.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  ])
+  signature = '0x' + signatureBuffer.toString('hex')
+  return signature
+}
+
+// For V4 Maket Maker Proxy (MMP)
+// Signature:
+// +------|---------|---------|---------|---------|------+
+// |  V   |    R    |    S    |userAddr |feeFactor| type |
+// +------|---------|---------|---------|---------|------+
```
Comment (kaichen) [inline src/signer/rfqv1.ts:46]: great!!! so clear to visualize data structure

File: src/handler/newOrder.ts (hunk 1 above)
Comment (tailingchen) [reply src/handler/newOrder.ts]: Fixed in d06dba6

File: src/handler/newOrder.ts (hunk 2 above)
Comment (tailingchen) [reply src/handler/newOrder.ts]: Fixed in 5b29dea

**Review by kaichen — APPROVED** [review]

File: src/types/index.ts [hunk 4]
```diff
@@ -100,3 +100,27 @@ export enum Protocol {
...
+  senderAddress: string
+  feeRecipientAddress: string
+  expirationTimeSeconds: string
+  exchangeAddress: string
+  feeFactor: number
+  salt: string
+  makerWalletSignature: string
+  quoteId: string
+  protocol: Protocol
+}
```
Comment (kaichen) [inline src/types/index.ts:126]: please leave some comments on fields, like some of them are out-of-date

File: src/handler/newOrder.ts [hunk 5]
```diff
@@ -22,16 +22,42 @@ import {
...
+  takerAssetAddress: string
+  expirationTimeSeconds: string
+  feeFactor: number
+  salt: string
+
+  // 0x protocol specific fields
+  makerAssetData: string
+  makerFee: string
+  takerAssetData: string
+  takerFee: string
```
Comment (kaichen) [inline src/handler/newOrder.ts:44]: `xxxFee` did not use from beginning

File: src/handler/newOrder.ts [hunk 6]
```diff
@@ -22,16 +22,42 @@ import {
...
+  senderAddress: string
+  feeRecipientAddress: string
```
Comment (kaichen) [inline src/handler/newOrder.ts:46]: on v5, we use this field as receiver address

File: src/handler/newOrder.ts [hunk 7]
```diff
@@ -22,16 +22,42 @@ import {
...
+
+  // Common fields
+  makerAddress: string
+  makerAssetAmount: string
+  makerAssetAddress: string
+  takerAddress: string
+  takerAssetAmount: string
+  takerAssetAddress: string
+  expirationTimeSeconds: string
+  feeFactor: number
```
Comment (kaichen) [inline src/handler/newOrder.ts:37]: tokenlon protocol field, work like BPS, should <= 10000

File: src/handler/newOrder.ts [hunk 8]
```diff
@@ -22,16 +22,42 @@ import {
 
 type NumberOrString = number | string
 
+interface Order {
+  quoteId: string
+  protocol: Protocol
```
Comment (kaichen) [inline src/handler/newOrder.ts:27]: order type as enum, [PMMV4, PMMV5, AMMV1, RFQ, AMMV2]

File: src/handler/newOrder.ts [hunk 9]
```diff
@@ -22,16 +22,42 @@ import {
 
 type NumberOrString = number | string
 
+interface Order {
+  quoteId: string
```
Comment (kaichen) [inline src/handler/newOrder.ts:26]: from market maker backend quoter

**Pull request merged**